src/server/controllers/vote.js

```diff
@@ -0,0 +1,140 @@
+const {body, validationResult} = require("express-validator");
+
+const Vote = require("../model/vote");
+// const http = require('http');
+// const url = require('url');
+
+//it return the number of user who have voted this post.
+exports.getVotesByEventId = function(req, res) {
+
+    // const queryObject = url.parse(req.url,true).query;
+    // console.log(queryObject);
+
+    //event id is accepted as a query param
+    var eventId = req.query.eventId;
+
+    Vote.find({"eventId": eventId}).exec(function(err, listVotes) {
+        if(err) {
+            console.log("Error: unable to get events");
+            res.status(400).end();
+        } else {
+
+            var upVote = 0;
+            var downVote = 0;
+
+            for (var i  in listVotes){
+                // console.log(listVotes[i]);
+                if(listVotes[i].vote == 1)
+                    upVote++;
+                else if(listVotes[i].vote == -1) //Condition not required
+                    downVote++;
+            }
+            res.json({"upVote": upVote, "downVote":downVote});
+        }
+    });
+};
+
+//Points of a user = No. of times he has upvoted or downvoted a post.
+exports.getPointsForAUser = function(req, res) {
+
+    //event id is accepted as a path param
+    var emailId = req.query.emailId;
+
+    // Comment.find({"eventId": req.body.eventId}).exec(function(err, listComments) {
+    Vote.find({"emailId": emailId}).exec(function(err, listVotes) {
+        if(err) {
+            console.log("Error: unable to get events");
+            res.status(400).end();
+        } else {
+            res.json({"points":listVotes.length});
+        }
+    });
+};
+
+
+//Get whether it's an upvote or downvote or No vote.
+//Points of a user = No. of times he has upvoted or downvoted a post.
+exports.getVoteByEventAndUser = function(req, res) {
+
+    var eventId = req.query.eventId;
+    var emailId = req.query.emailId;
+
+    Vote.find({"eventId": eventId, "emailId":emailId}).exec(function(err, singleVote) {
+        if(err) {
+            console.log("Error: unable to get events");
+            res.status(400).end();
+        } else {
+            //there will always be a single vote
+            var response = "";
```

Comment:
If you look at coveralls not all paths are covered. It's only hitting downvote case.